### PR TITLE
feat: [CI-5326]: add basic pill radio button css

### PR DESCRIPTION
**Pull request by d1wilko**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 1]
```diff
@@ -5,6 +5,32 @@
...
+  background-color: white !important;
+  border: 1px solid lightgray;
+  cursor: pointer;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.05);
+  margin-right: 0.5rem !important;
+}
+
+.asPill input {
+  visibility: hidden;
+}
```
Comment (knagurski) [inline packages/uicore/src/components/RadioButton/RadioButton.css:22]: The input is already hidden... you can probably remove this

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 2]
```diff
@@ -5,6 +5,32 @@
...
+
+.asPill span {
+  margin: 0;
+}
+
+.pillChecked {
+  background-color: #4090de !important;
+  border-color: #3b7fd1;
```
Comment (knagurski) [inline packages/uicore/src/components/RadioButton/RadioButton.css:30]: Let's grab these from the existing colour custom properties

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 3]
```diff
@@ -5,6 +5,32 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+.asPill {
+  font-family: sans-serif;
+  display: inline-block;
+  padding: 0.66rem 1.5rem !important;
+  border-radius: 2rem;
```
Comment (knagurski) [inline packages/uicore/src/components/RadioButton/RadioButton.css:12]: Probably best to use `em`s so we can grow/shrink this with the font-size

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 4]
```diff
@@ -5,6 +5,32 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+.asPill {
+  font-family: sans-serif;
```
Comment (knagurski) [inline packages/uicore/src/components/RadioButton/RadioButton.css:9]: Probably don't need this

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 5]
```diff
@@ -5,6 +5,31 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+.asPill {
+  display: inline-block;
+  padding: 0.66rem 1.5rem !important;
+  border-radius: 2em;
+  background-color: white !important;
```
Comment (crystal-fong) [inline packages/uicore/src/components/RadioButton/RadioButton.css:12]: could this use the `var(--white)` as well?

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 6]
```diff
@@ -5,6 +5,31 @@
...
+}
+
+.asPill span {
+  margin: 0;
+}
+
+.pillChecked {
+  background-color: var(--primary-6) !important;
+  border-color: var(--primary-7);
+  color: white !important;
```
Comment (crystal-fong) [inline packages/uicore/src/components/RadioButton/RadioButton.css:26]: same as above

File: packages/uicore/src/components/RadioButton/RadioButton.css [hunk 7]
```diff
@@ -5,6 +5,31 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
+.asPill {
+  display: inline-block;
+  padding: 0.66rem 1.5rem !important;
+  border-radius: 2em;
+  background-color: white !important;
+  border: 1px solid lightgray;
```
Comment (crystal-fong) [inline packages/uicore/src/components/RadioButton/RadioButton.css:13]: I think we should have a gray variable that can replace this?

**Review by crystal-fong — APPROVED** [review]
left some small nitpicks but otherwise good to go

File: packages/uicore/src/components/RadioButton/RadioButton.tsx [hunk 8]
```diff
@@ -30,11 +31,19 @@ export function RadioButton({
...
   onChange = () => undefined,
   ...props
 }: RadioButtonProps & StyledProps): ReactElement {
   return (
-    <label className={styledClasses(props, css.radio, className)} {...omitStyledProps(props)}>
+    <label
+      className={styledClasses(
+        props,
+        asPill ? css.asPill : css.radio,
+        asPill && checked ? css.pillChecked : '',
```
Comment (bgribben-harness) [inline packages/uicore/src/components/RadioButton/RadioButton.tsx:43]: This looks a bit cumbersome to me - it'd be neat if we could extract out a `type` variable to remove the inline `asPill ? css.asPill : css.radio` ternary, and make use of the input's `:checked` state to add the `pillChecked` styles, but I'm not sure how difficult that is to do - might be handiest to leave it all as-is!

**Review by bgribben-harness — APPROVED** [review]
LGTM

File: packages/uicore/src/components/RadioButton/RadioButton.tsx (hunk 8 above)
Comment (knagurski) [reply packages/uicore/src/components/RadioButton/RadioButton.tsx:43]: When `has` is available we can move all this to CSS using `.asPill:has(input:checked)`, which would be awesome!

**Pull request merged**